src/gwt/src/org/rstudio/studio/client/workbench/views/environment/EnvironmentPane.java

```diff
@@ -415,8 +415,17 @@ private void executeFunctionForObject(String function, String objectName)
    private Widget createImportMenu()
    {
       ToolbarPopupMenu menu = new ToolbarPopupMenu();
-      menu.addItem(commands_.importDatasetFromFile().createMenuItem(false));
-      menu.addItem(commands_.importDatasetFromURL().createMenuItem(false));
+      
+      if (prefs_.useDataImport().getValue())
+      {
+         menu.addItem(commands_.importDatasetFromCSV().createMenuItem(false));
```

Comment:
You'll want to set the visibility of the commands instead of conditionally adding them to the menu, since these commands also appear on the top-level app menu and need to be shown/hidden there too.